_pytest/logging.py

```diff
@@ -153,7 +153,7 @@ def catching_logs(handler, formatter=None, level=None):
         root_logger.addHandler(handler)
     if level is not None:
         orig_level = root_logger.level
-        root_logger.setLevel(level)
+        root_logger.setLevel(min(orig_level, level))
```

Comment:
i believe we need to ensure the level of the handler is saved/restored as well, but set to the exact level given,
so the handling will happen at the expected level for the designated handler, even if other handlers work with the original level

Reply:
I believe setting the log capturing handlers level here
https://github.com/pytest-dev/pytest/blob/b2b629f46230b5ec6b87a4cefde3bfef1875984e/_pytest/logging.py#L146
 is enough, or correct me if I am wrong.

Reply:
correct, that's more than sufficient, thanks for looking that up :+1: